tests/test_user_auth.py

```diff
@@ -0,0 +1,102 @@
+                     'email': 'user@test.com',
+                     'password': '1234'
+                     }
+        response = self.data.post(self.REGISTER_URL, json.dumps(self.data),
+                                  content_type="application/json")
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual("You have successfully registered. Kindly log in",
+                         str(response.data))
+
+    def test_user_registration_when_user_already_exists(self):
+        """Test registration of an already existing user."""
+        self.data = {'username': 'test_username',
+                     'email': 'user@test.com',
+                     'password': '1234'
+                     }
+        self.data.post(self.REGISTER_URL, json.dumps(self.data),
+                       content_type="application/json")
+        response = self.data.post(self.REGISTER_URL, json.dumps(self.data),
+                                  content_type="application/json")
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual("User already exists!", str(response.data))
+
+    def test_user_registration_with_no_email(self):
+        """Test for user registration with no email."""
+        self.data = {'username': 'test_username',
+                     'password': '1234'
+                     }
+        response = self.data.post(self.REGISTER_URL, json.dumps(self.data),
+                                  content_type="application/json")
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual("Please provide an email!", str(response.data))
+
+    def test_user_registration_username_already_exists(self):
+        """Test for registration with an already existing username."""
+        self.data = {'username': 'test_username',
+                     'email': 'user@test.com',
+                     'password': '1234'
+                     }
+        self.data.post(self.REGISTER_URL, json.dumps(self.data),
+                       content_type="application/json")
+        self.data = {'username': 'test_username',
+                     'email': 'me@user.com',
+                     'password': '56789'
+                     }
+        response = self.data.post(self.REGISTER_URL, json.dumps(self.data),
+                                  content_type="application/json")
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual("Username already exists! Please provide another",
+                         str(response.data))
+
+    def test_user_login(self):
+        """Test for successful user login."""
+        self.data = {'username': "johndoe",
+                     'email': "johndoe@gmail.com"
+                     }
+        response = self.data.post(self.LOGIN_URL, json.dumps(self.data),
+                                  content_type="application/json")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual("You have successfully logged in",
```

Comment:
A better test for login would be to check that an access token has been sent back